### PR TITLE
feat(list, list-item): add non-interactive option to remove hover/pointer changes.

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [#8634](https://github.com/Esri/calcite-design-system/issues/8634)

## Summary

- Add `interaction-mode="interactive" (default) | "static"` prop.
- Setting `interaction-mode="static"` and `selection-appearance="icon"` removes hover/pointer changes from main content area.
- Setting `interaction-mode="static"` and `selection-appearance="border"` falls back to interactive hover/pointer changes and logs console warning message.
- Setting `interaction-mode="static"` and `selection-mode="none"` removes hover/pointer changes.

### CODE REVIEW

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (driskull) [inline packages/calcite-components/src/components/list-item/list-item.tsx:228]: I think this would need to be called on `componentWillLoad` as well, otherwise it would only warn after one of these was changed.

Comment (driskull) [reply packages/calcite-components/src/components/list-item/list-item.tsx:228]: Should this warning occur on the list instead of the list item? The property on the list-item is internal so its weird to log here.

**Review by driskull — CHANGES_REQUESTED** [review]
Can we move the warning log to the list?

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (aPreciado88) [reply packages/calcite-components/src/components/list-item/list-item.tsx]: I moved this logic into list.tsx 🚀

**Review by driskull — APPROVED** [review]
Code looks good! can we just not reflect the internal prop?

File: packages/calcite-components/src/components/list/list.tsx
Comment (aPreciado88) [reply packages/calcite-components/src/components/list/list.tsx:990]: Aaron, Franco and me went over the issue, and landed on setting a detailed JSDoc for documentation while also logging the warning message. But maybe this can be reconsidered? 
cc @jcfranco @ashetland

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (aPreciado88) [reply packages/calcite-components/src/components/list-item/list-item.tsx]: Thanks for catching this! I updated it to not be reflected.

**Comment by aPreciado88** [comment]
@ashetland @SkyeSeitz Can you please take a look at the new snapshots?

File: packages/calcite-components/src/components/list/list.tsx
Comment (ashetland) [reply packages/calcite-components/src/components/list/list.tsx:990]: I defer to @jcfranco and @driskull.

Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.tsx:990]: Great question. Since this is something teams have been asking for, let’s keep the warning in place to ensure developers use the proper configuration. After #10310, we can explore showing this and related warnings (e.g., [color-picker’s incompatible value + format mode](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/components/color-picker/color-picker.tsx#L1133)) only in dev builds, as suggested in [#10038](https://github.com/Esri/calcite-design-system/issues/10038).

File: packages/calcite-components/src/components/list/list.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.tsx:231]: +@geospatialem @DitwanP for editorial review. 📖👀

File: packages/calcite-components/src/components/list/list.stories.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.stories.ts:1154]: Can you merge these stories into one with a descriptor for each use case (e.g., [slider fill placements](https://github.com/Esri/calcite-design-system/blob/4c2c623d59441be8ddc66866748b7b104810b5ac/packages/calcite-components/src/components/slider/slider.stories.ts#L466-L524))?

File: packages/calcite-components/src/components/list/list.stories.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/list/list.stories.ts:1155]: Nitpick: is setting `width` necessary for this test?

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/list-item/list-item.tsx:699]: This naming a bit misleading. Since `containerInteractive` and `containerNotInteractive` are opposites, this ends up reading as `containerInteractive || containerInteractive`.

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/list-item/list-item.tsx:678]: I _think_ this can be simplified to:

```tsx
const containerInteractive =
  interactionMode === "interactive" ||
  (
    interactionMode === "static" &&
    selectionMode !== "none" &&
    selectionAppearance === "border"
  );

// ...

[CSS.containerHover]: containerInteractive,
```

**Review by jcfranco — APPROVED** [review]
Good to merge after addressing feedback. 

👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍
👍✅👍👍👍✅👍👍✅✅👍👍✅✅✅👍👍✅✅✅👍✅✅✅✅👍✅👍
👍✅✅👍👍✅👍✅👍👍✅👍👍✅👍👍✅👍👍👍👍✅👍👍👍👍✅👍
👍✅👍✅👍✅👍✅👍👍✅👍👍✅👍👍✅👍👍👍👍✅✅✅👍👍✅👍
👍✅👍👍✅✅👍✅👍👍✅👍👍✅👍👍✅👍👍👍👍✅👍👍👍👍👍👍
👍✅👍👍👍✅👍👍✅✅👍👍✅✅✅👍👍✅✅✅👍✅✅✅✅👍✅👍
👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍

File: packages/calcite-components/src/components/list/list.stories.ts
Comment (aPreciado88) [reply packages/calcite-components/src/components/list/list.stories.ts:1155]: This is not necessary, I can take the width out.

File: packages/calcite-components/src/components/list-item/list-item.tsx
Comment (aPreciado88) [reply packages/calcite-components/src/components/list-item/list-item.tsx:678]: Thanks for catching that, this update works!

**Comment by ashetland** [comment]
Lookin' good!

**Review by ashetland — APPROVED** [review]

File: packages/calcite-components/src/components/list/list.tsx
Comment (geospatialem) [reply packages/calcite-components/src/components/list/list.tsx:231]: @aPreciado88 Can you speak more to the "why" behind not using the props together? Maybe something we can also add to enhance the component's documentation page. I didn't see context in the PR's summary or in the console warning, so wanted to be sure I understand the "why" a bit more before providing a review. Did observe different double quotes, so recommend changing those in the interim.

```suggestion
   * Do not combine `interaction-mode="static"` and `selection-appearance="border"`
```

Comment (aPreciado88) [reply packages/calcite-components/src/components/list/list.tsx:231]: The reason for the console warning and not using the props together is described in this [issue](https://github.com/Esri/calcite-design-system/issues/8634). Under Calcite Design Solution > selection-appearance=“border”.

Comment (geospatialem) [reply packages/calcite-components/src/components/list/list.tsx:231]: Summarizing our discussions via chat:
- We'll enhance the above to mention it is recommended to only add the `interaction-mode="static"` and `"selection-appearance="none"` together (_see suggestion below_)
- We'll provide more context via some images to the `list`'s doc page under a new "Recommendations" section cc: @ashetland @jcfranco 

```suggestion
   * The `"static"` value should only be used when `selectionAppearance` is `"none"`.
```

Comment (ashetland) [reply packages/calcite-components/src/components/list/list.tsx:231]: The property should be `selection-mode`, not `selection-appearance`.

Comment (aPreciado88) [reply packages/calcite-components/src/components/list/list.tsx]: Updated the PR with the requested changes. Thanks for catching that @ashetland!

**Pull request merged**
